json_journal/data_analysis.py

```diff
@@ -1,21 +1,50 @@
+'''Functions for various analysis on data'''
+
+
 def calculate_balance(ledger):
-    all_entries = {}
+    '''
+    Calculate balance for all entries.
+
+    Parameters
+    ----------
+    Ledger : dict
+        Loaded json file.
+
+    Returns
+    -------
+    All_balances : dict
+        Contains full account names and balances for each account.
+    '''
+    all_balances = {}
     for transaction in ledger:
         entries = transaction['entries']
 
         for entry in entries:
             amount = entry['amount']
             account = entry['account']
 
-            if account in all_entries:
-                new_amount = amount + all_entries.get(account)
-                all_entries[account] = new_amount
+            if account in all_balances:
+                new_amount = amount + all_balances.get(account)
+                all_balances[account] = new_amount
             else:
-                all_entries[account] = amount
-    return all_entries
+                all_balances[account] = amount
+    return all_balances
 
 
 def parent_account_balances(full_account_name_balances):
+    '''
+    Calculate balances for all parent accounts.
+
+    Parameters
+    ----------
+    Full_account_name_balances : dict
+        Contains the full account name and corresponding balance.
+
+    Returns
+    -------
+    New_balances : dict
+        Contains balances for each parent account.
+    '''
     new_balances = {}
     for full_account_name in full_account_name_balances.keys():
         account_separated = full_account_name.split(':')
```

Comment:
This colon could be defined as a constant, since it's used in multiple places and is likely to be used again. Note convention is to define constants in `SCREAMING_SNAKE_CASE`

Reply:
done